informational/req-res-reliability.md

```diff
@@ -0,0 +1,62 @@
+---
+title: REQ-RES-RELIABILITY
+name: Request-response protocols reliability
+category: Best Current Practice
+tags: [informational]
+editor: Oleksandr Kozlov <oleksandr@status.im>
+contributors:
+- Prem Chaitanya Prathi <prem@status.im>
+- Danish Arora <danish@status.im>
+---
+
+## Abstract
+This RFC describes set of instructions used across different [WAKU2](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/10/waku2.md?plain=1#L3) implementations for improved reliability in request-response protocols such as [WAKU2-LIGHTPUSH](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/19/lightpush.md?plain=1#L3C11-L3C26) and [WAKU2-FILTER](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/12/filter.md?plain=1#L3).
+
+## Motivation
+
+Descriptions of mentioned protocols do not define some of the real world use cases that are oftenly observed in unreliable network environment. Such use cases can be: recovery from offline state, decrease rate of missed messages, increase probability of messages being broadcasted within the network.
+
+## Suggestions
+
+### Node health
+
+As a useful metric to define and implement for determining quality of provided service by a node:
+- unhealthy - no peer connections are available regardless of protocol;
+- minimally healthy:
+  - Relay has less than 4 peers connected;
+  - Filter and LightPush has one per each peer connection available;
+- sufficiently healthy:
+  - Relay has minimum 4 peers connected;
+  - more than 1 connection in Filter and at least 2 connections available in LightPush;
+
+### Peers and connection management
+
+- Each protocols should retain a pool of reliable peers. In case a protocol failed to use any peer more than once - connection to it should be dropped and new peer should be added to the pool instead. 
```

Comment:
Rather than calling it protocol, we should indicate that each client should retain a pool of peers for Filter and lightPush. 

Also i am wondering if we have an understanding of what i means by using a peer. We are looking at a peer-to-peer network where serviceNodes availability is not gauranteed unless they are fleet nodes. e.g: status-desktop instances may not be always running. 
There could be different scenarios why connectivity to a specific peer fails at specified time such as rate limits reached or ip-colocation-limit hit . That doesn't necessarily mean it is a bad peer. Hopefully we are not referring to descoring the peer.

Reply:
added definition of failure for each protocol 

`descoring` is out of scope and we were considering something like that for js-waku in https://github.com/waku-org/js-waku/issues/2017